### PR TITLE
Modify no_merge_train_val to retrain.

### PR DESCRIPTION
**Pull request by Gordon119**
## What does this PR do?

As title, the `no_merge_train_val` in `search_params.py` can be confused with the `merge_train_val` in `torch_trainer.py`. Although they both indicate merging training and validation set, the former one is used for retraining. This PR modify the name of it.

## Test CLI & API (`bash tests/autotest.sh`)
Test APIs used by main.py.
- [ ] Test Pass
  - (Copy and paste the last outputted line here.)
- [x] Not Applicable (i.e., the PR does not include API changes.)

## Check API Document
If any new APIs are added, please check if the description of the APIs is added to API document. 
- [ ] API document is updated ([linear](https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/api/linear.html), [nn](https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/api/nn.html))
- [x] Not Applicable (i.e., the PR does not include API changes.)

## Test quickstart & API (`bash tests/docs/test_changed_document.sh`)
If any APIs in quickstarts or tutorials are modified, please run this test to check if the current examples can run correctly after the modified APIs are released.

### CODE REVIEW

**Review by cjlin1 — APPROVED** [review]

File: search_params.py [hunk 1]
```diff
@@ -165,15 +165,15 @@ def prepare_retrain_config(best_config, best_log_dir, merge_train_val):
         optimal_idx = log_metric.argmax() if best_config.mode == "max" else log_metric.argmin()
         best_config.epochs = optimal_idx.item() + 1  # plus 1 for epochs
     else:
-        best_config.merge_train_val = False
+        best_config.retrain = False
```
Comment (Eleven1Liu) [inline search_params.py:168]: Is `best_config.merge_train_val` used in main.py? ([link](https://github.com/ASUS-AICS/LibMultiLabel/blob/f325f4877c91751c2075307eb2b7bc486892e5e2/main.py#L48))

**Review by Eleven1Liu — CHANGES_REQUESTED** [review]
Please check when to use `config.merge_train_val` and when to use `args.retrain`.

**Review by Eleven1Liu — APPROVED** [review]
